# Assignment 1

## Data Wrangling I 

Perform the following operations using Python on any open-source dataset (e.g., data.csv) 

Import all the required Python Libraries. 

Locate an open-source data from the web (e.g. https://www.kaggle.com). Provide a clear description of the data and its source (i.e., URL of the web site). 

1. Load the Dataset into pandas’ data frame. 

2. Data Preprocessing: check for missing values in the data using pandas insult (), describe() function to get some initial statistics. Provide variable descriptions. Types of variables etc. Check the dimensions of the data frame. 

3. Data Formatting and Data Normalization: Summarize the types of variables by checking the data types (i.e., character, numeric, integer, factor, and logical) of the variables in the data set. If variables are not in the correct data type, apply proper type conversions. 

4. Turn categorical variables into quantitative variables in Python. 

In addition to the codes and outputs, explain every operation that you do in the above steps and explain everything that you do to import/read/scrape the data set.

In [148]:
import pandas as pd
import numpy as np

# 1. Load the Dataset into pandas’ data frame. 

## Read CSV data using pandas

In [177]:
FILE_PATH='./datasets/melb_data.csv'
raw_data=pd.read_csv(FILE_PATH)
raw_data.head()

Suburb           Address  Rooms Type      Price Method SellerG  \
0  Abbotsford      85 Turner St      2    h  1480000.0      S  Biggin   
1  Abbotsford   25 Bloomburg St      2    h  1035000.0      S  Biggin   
2  Abbotsford      5 Charles St      3    h  1465000.0     SP  Biggin   
3  Abbotsford  40 Federation La      3    h   850000.0     PI  Biggin   
4  Abbotsford       55a Park St      4    h  1600000.0     VB  Nelson   

        Date  Distance  Postcode  ...  Bathroom  Car  Landsize  BuildingArea  \
0  3/12/2016       2.5    3067.0  ...       1.0  1.0     202.0           NaN   
1  4/02/2016       2.5    3067.0  ...       1.0  0.0     156.0          79.0   
2  4/03/2017       2.5    3067.0  ...       2.0  0.0     134.0         150.0   
3  4/03/2017       2.5    3067.0  ...       2.0  1.0      94.0           NaN   
4  4/06/2016       2.5    3067.0  ...       1.0  2.0     120.0         142.0   

   YearBuilt  CouncilArea Lattitude  Longtitude             Regionname  \
0        NaN        Yarra  -37.7996    144.9984  Northern Metropolitan   
1     1900.0        Yarra  -37.8079    144.9934  Northern Metropolitan   
2     1900.0        Yarra  -37.8093    144.9944  Northern Metropolitan   
3        NaN        Yarra  -37.7969    144.9969  Northern Metropolitan   
4     2014.0        Yarra  -37.8072    144.9941  Northern Metropolitan   

  Propertycount  
0        4019.0  
1        4019.0  
2        4019.0  
3        4019.0  
4        4019.0  

[5 rows x 21 columns]

In [208]:
data=raw_data.copy(deep=True)

# Data description

## Background:
Suburb: Suburb

Address: Address

Rooms: Number of rooms

Price: Price in Australian dollars

Method:
S - property sold;
SP - property sold prior;
PI - property passed in;
PN - sold prior not disclosed;
SN - sold not disclosed;
NB - no bid;
VB - vendor bid;
W - withdrawn prior to auction;
SA - sold after auction;
SS - sold after auction price not disclosed.
N/A - price or highest bid not available.

Type:
br - bedroom(s);
h - house,cottage,villa, semi,terrace;
u - unit, duplex;
t - townhouse;
dev site - development site;
o res - other residential.

SellerG: Real Estate Agent

Date: Date sold

Distance: Distance from CBD in Kilometres

Regionname: General Region (West, North West, North, North east …etc)

Propertycount: Number of properties that exist in the suburb.

Bedroom2 : Scraped # of Bedrooms (from different source)

Bathroom: Number of Bathrooms

Car: Number of carspots

Landsize: Land Size in Metres

BuildingArea: Building Size in Metres

YearBuilt: Year the house was built

CouncilArea: Governing council for the area

Lattitude: Self explanitory

Longtitude: Self explanitory

In [209]:
data.shape

(13580, 21)

### Dataset size:
- No of Rows = 13580
- No of Columns = 21

In [210]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13580 entries, 0 to 13579
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Suburb         13580 non-null  object 
 1   Address        13580 non-null  object 
 2   Rooms          13580 non-null  int64  
 3   Type           13580 non-null  object 
 4   Price          13580 non-null  float64
 5   Method         13580 non-null  object 
 6   SellerG        13580 non-null  object 
 7   Date           13580 non-null  object 
 8   Distance       13580 non-null  float64
 9   Postcode       13580 non-null  float64
 10  Bedroom2       13580 non-null  float64
 11  Bathroom       13580 non-null  float64
 12  Car            13518 non-null  float64
 13  Landsize       13580 non-null  float64
 14  BuildingArea   7130 non-null   float64
 15  YearBuilt      8205 non-null   float64
 16  CouncilArea    12211 non-null  object 
 17  Lattitude      13580 non-null  float64
 18  Longti

In [211]:
data.dtypes.value_counts()

float64    12
object      8
int64       1
dtype: int64

# 2. Data Preprocessing: check for missing values in the data using pandas insult (), describe() function to get some initial statistics. Provide variable descriptions. Types of variables etc. Check the dimensions of the data frame.

## Check for null/nan values

In [181]:
data.isnull().sum()

Suburb              0
Address             0
Rooms               0
Type                0
Price               0
Method              0
SellerG             0
Date                0
Distance            0
Postcode            0
Bedroom2            0
Bathroom            0
Car                62
Landsize            0
BuildingArea     6450
YearBuilt        5375
CouncilArea      1369
Lattitude           0
Longtitude          0
Regionname          0
Propertycount       0
dtype: int64

In [182]:
data.isnull().sum().sum()

13256

### Total null values = 13256
- Car              -  62
- BuildingArea  -   6450
- YearBuilt     -  5375
- CouncilArea   -  1369

In [183]:
# Car - dtpye - float64
# Car - Number of carspots
mode_no=data['Car'].mode()
print('Mode no of car spots = ',mode_no)
print('Total no of NaN values = ',data['Car'].isna().sum())
# Replace NaN values by mean value
data['Car']=data['Car'].fillna(mode_no[0])
print('Total no of NaN values after replacement = ',data['Car'].isna().sum())

Mode no of car spots =  0    2.0
dtype: float64
Total no of NaN values =  62
Total no of NaN values after replacement =  0


In [184]:
# BuildingArea - Building Size in Metres
# BuildingArea - dtype - float64
avg_no=data['BuildingArea'].mean()
print('Average value of BuildingArea = ',avg_no)
print('Total no of NaN values = ',data['BuildingArea'].isna().sum())
# Replace NaN values by mean value
data['BuildingArea']=data['BuildingArea'].fillna(avg_no)
print('Total no of NaN values after replacement = ',data['BuildingArea'].isna().sum())

Average value of BuildingArea =  151.96764988779805
Total no of NaN values =  6450
Total no of NaN values after replacement =  0


In [185]:
# CouncilArea -  Governing council for the area
# CouncilArea - dtype - object
data['CouncilArea'].value_counts()

Moreland             1163
Boroondara           1160
Moonee Valley         997
Darebin               934
Glen Eira             848
Stonnington           719
Maribyrnong           692
Yarra                 647
Port Phillip          628
Banyule               594
Bayside               489
Melbourne             470
Hobsons Bay           434
Brimbank              424
Monash                333
Manningham            311
Whitehorse            304
Kingston              207
Whittlesea            167
Hume                  164
Wyndham                86
Knox                   80
Maroondah              80
Melton                 66
Frankston              53
Greater Dandenong      52
Casey                  38
Nillumbik              36
Yarra Ranges           18
Cardinia                8
Macedon Ranges          7
Unavailable             1
Moorabool               1
Name: CouncilArea, dtype: int64

In [186]:
print('Total no of NaN values = ',data['CouncilArea'].isna().sum())
# Drop rows containing nan values
data=data.dropna(subset=['CouncilArea'],axis=0,how='any')
print('Total no of NaN values after dropping rows = ',data['CouncilArea'].isna().sum())

Total no of NaN values =  1369
Total no of NaN values after dropping rows =  0


In [187]:
# YearBuilt - Year the house was built
# YearBuilt - dtype - float64
print('Total no of NaN values = ',data['YearBuilt'].isna().sum())
# Drop rows containing NaN values
data=data.dropna(subset=['YearBuilt'],axis=0,how='any')
print('Total no of NaN values after replacement = ',data['YearBuilt'].isna().sum())

Total no of NaN values =  4763
Total no of NaN values after replacement =  0


In [188]:
data.head()

Suburb          Address  Rooms Type      Price Method SellerG  \
1  Abbotsford  25 Bloomburg St      2    h  1035000.0      S  Biggin   
2  Abbotsford     5 Charles St      3    h  1465000.0     SP  Biggin   
4  Abbotsford      55a Park St      4    h  1600000.0     VB  Nelson   
6  Abbotsford     124 Yarra St      3    h  1876000.0      S  Nelson   
7  Abbotsford    98 Charles St      2    h  1636000.0      S  Nelson   

        Date  Distance  Postcode  ...  Bathroom  Car  Landsize  BuildingArea  \
1  4/02/2016       2.5    3067.0  ...       1.0  0.0     156.0          79.0   
2  4/03/2017       2.5    3067.0  ...       2.0  0.0     134.0         150.0   
4  4/06/2016       2.5    3067.0  ...       1.0  2.0     120.0         142.0   
6  7/05/2016       2.5    3067.0  ...       2.0  0.0     245.0         210.0   
7  8/10/2016       2.5    3067.0  ...       1.0  2.0     256.0         107.0   

   YearBuilt  CouncilArea Lattitude  Longtitude             Regionname  \
1     1900.0        Yarra  -37.8079    144.9934  Northern Metropolitan   
2     1900.0        Yarra  -37.8093    144.9944  Northern Metropolitan   
4     2014.0        Yarra  -37.8072    144.9941  Northern Metropolitan   
6     1910.0        Yarra  -37.8024    144.9993  Northern Metropolitan   
7     1890.0        Yarra  -37.8060    144.9954  Northern Metropolitan   

  Propertycount  
1        4019.0  
2        4019.0  
4        4019.0  
6        4019.0  
7        4019.0  

[5 rows x 21 columns]

In [189]:
data.shape

(7448, 21)

## Initial statistics

In [190]:
data.describe()

Rooms         Price     Distance     Postcode     Bedroom2  \
count  7448.000000  7.448000e+03  7448.000000  7448.000000  7448.000000   
mean      2.888695  1.053877e+06     9.658526  3103.311359     2.862648   
std       0.980317  6.658239e+05     5.569200    86.528293     0.975632   
min       1.000000  8.500000e+04     0.000000  3000.000000     0.000000   
25%       2.000000  6.104375e+05     5.900000  3044.000000     2.000000   
50%       3.000000  8.675000e+05     9.000000  3084.000000     3.000000   
75%       3.000000  1.305000e+06    12.400000  3147.000000     3.000000   
max      10.000000  9.000000e+06    47.400000  3977.000000    10.000000   

          Bathroom          Car      Landsize  BuildingArea    YearBuilt  \
count  7448.000000  7448.000000   7448.000000   7448.000000  7448.000000   
mean      1.556928     1.559211    478.352444    143.316705  1964.605666   
std       0.701421     0.924132    982.728391     82.939290    37.472441   
min       0.000000     0.000000      0.000000      0.000000  1196.000000   
25%       1.000000     1.000000    144.750000     98.000000  1940.000000   
50%       1.000000     1.000000    354.000000    139.000000  1970.000000   
75%       2.000000     2.000000    628.000000    157.951400  2000.000000   
max       8.000000    10.000000  37000.000000   3112.000000  2018.000000   

         Lattitude   Longtitude  Propertycount  
count  7448.000000  7448.000000    7448.000000  
mean    -37.809109   144.991628    7482.585526  
std       0.075412     0.097721    4344.438850  
min     -38.164920   144.542370     389.000000  
25%     -37.855605   144.930017    4442.000000  
50%     -37.804200   144.997100    6567.000000  
75%     -37.759175   145.052700   10331.000000  
max     -37.457090   145.526350   21650.000000

In [191]:
data.describe(include='object')

Suburb           Address  Type Method SellerG        Date  \
count        7448              7448  7448   7448    7448        7448   
unique        293              7353     3      5     216          51   
top     Reservoir  36 Aberfeldie St     h      S  Nelson  27/05/2017   
freq          187                 3  4727   4780     881         276   

       CouncilArea             Regionname  
count         7448                   7448  
unique          31                      8  
top       Moreland  Southern Metropolitan  
freq           748                   2711

## 3. Data Formatting and Data Normalization: Summarize the types of variables by checking the data types (i.e., character, numeric, integer, factor, and logical) of the variables in the data set. If variables are not in the correct data type, apply proper type conversions. 


In [192]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7448 entries, 1 to 12212
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Suburb         7448 non-null   object 
 1   Address        7448 non-null   object 
 2   Rooms          7448 non-null   int64  
 3   Type           7448 non-null   object 
 4   Price          7448 non-null   float64
 5   Method         7448 non-null   object 
 6   SellerG        7448 non-null   object 
 7   Date           7448 non-null   object 
 8   Distance       7448 non-null   float64
 9   Postcode       7448 non-null   float64
 10  Bedroom2       7448 non-null   float64
 11  Bathroom       7448 non-null   float64
 12  Car            7448 non-null   float64
 13  Landsize       7448 non-null   float64
 14  BuildingArea   7448 non-null   float64
 15  YearBuilt      7448 non-null   float64
 16  CouncilArea    7448 non-null   object 
 17  Lattitude      7448 non-null   float64
 18  Longtit

In [193]:
data.select_dtypes('float64').head(10)

Price  Distance  Postcode  Bedroom2  Bathroom  Car  Landsize  \
1   1035000.0       2.5    3067.0       2.0       1.0  0.0     156.0   
2   1465000.0       2.5    3067.0       3.0       2.0  0.0     134.0   
4   1600000.0       2.5    3067.0       3.0       1.0  2.0     120.0   
6   1876000.0       2.5    3067.0       4.0       2.0  0.0     245.0   
7   1636000.0       2.5    3067.0       2.0       1.0  2.0     256.0   
9   1097000.0       2.5    3067.0       3.0       1.0  2.0     220.0   
11  1350000.0       2.5    3067.0       3.0       2.0  2.0     214.0   
12   750000.0       2.5    3067.0       2.0       2.0  1.0       0.0   
15  1310000.0       2.5    3067.0       2.0       1.0  2.0     238.0   
16  1200000.0       2.5    3067.0       3.0       2.0  1.0     113.0   

    BuildingArea  YearBuilt  Lattitude  Longtitude  Propertycount  
1           79.0     1900.0   -37.8079    144.9934         4019.0  
2          150.0     1900.0   -37.8093    144.9944         4019.0  
4          142.0     2014.0   -37.8072    144.9941         4019.0  
6          210.0     1910.0   -37.8024    144.9993         4019.0  
7          107.0     1890.0   -37.8060    144.9954         4019.0  
9           75.0     1900.0   -37.8010    144.9989         4019.0  
11         190.0     2005.0   -37.8085    144.9964         4019.0  
12          94.0     2009.0   -37.8078    144.9965         4019.0  
15          97.0     1890.0   -37.8090    144.9976         4019.0  
16         110.0     1880.0   -37.8056    144.9930         4019.0

In [195]:
## Convert datatype of columns Postcode,Bedroom2,Bathroom,Car,YearBuilt into int64
data=data.astype({'Postcode':'int64','Bedroom2':'int64','Bathroom':'int64','Car':'int64','YearBuilt':'int64'})

In [196]:
data.head(10)

Suburb              Address  Rooms Type      Price Method SellerG  \
1   Abbotsford      25 Bloomburg St      2    h  1035000.0      S  Biggin   
2   Abbotsford         5 Charles St      3    h  1465000.0     SP  Biggin   
4   Abbotsford          55a Park St      4    h  1600000.0     VB  Nelson   
6   Abbotsford         124 Yarra St      3    h  1876000.0      S  Nelson   
7   Abbotsford        98 Charles St      2    h  1636000.0      S  Nelson   
9   Abbotsford        10 Valiant St      2    h  1097000.0      S  Biggin   
11  Abbotsford      40 Nicholson St      3    h  1350000.0     VB  Nelson   
12  Abbotsford  123/56 Nicholson St      2    u   750000.0      S  Biggin   
15  Abbotsford        16 William St      2    h  1310000.0      S  Jellis   
16  Abbotsford          42 Henry St      3    h  1200000.0      S  Jellis   

          Date  Distance  Postcode  ...  Bathroom  Car  Landsize  \
1    4/02/2016       2.5      3067  ...         1    0     156.0   
2    4/03/2017       2.5      3067  ...         2    0     134.0   
4    4/06/2016       2.5      3067  ...         1    2     120.0   
6    7/05/2016       2.5      3067  ...         2    0     245.0   
7    8/10/2016       2.5      3067  ...         1    2     256.0   
9    8/10/2016       2.5      3067  ...         1    2     220.0   
11  12/11/2016       2.5      3067  ...         2    2     214.0   
12  12/11/2016       2.5      3067  ...         2    1       0.0   
15  15/10/2016       2.5      3067  ...         1    2     238.0   
16  16/07/2016       2.5      3067  ...         2    1     113.0   

    BuildingArea  YearBuilt  CouncilArea Lattitude  Longtitude  \
1           79.0       1900        Yarra  -37.8079    144.9934   
2          150.0       1900        Yarra  -37.8093    144.9944   
4          142.0       2014        Yarra  -37.8072    144.9941   
6          210.0       1910        Yarra  -37.8024    144.9993   
7          107.0       1890        Yarra  -37.8060    144.9954   
9           75.0       1900        Yarra  -37.8010    144.9989   
11         190.0       2005        Yarra  -37.8085    144.9964   
12          94.0       2009        Yarra  -37.8078    144.9965   
15          97.0       1890        Yarra  -37.8090    144.9976   
16         110.0       1880        Yarra  -37.8056    144.9930   

               Regionname Propertycount  
1   Northern Metropolitan        4019.0  
2   Northern Metropolitan        4019.0  
4   Northern Metropolitan        4019.0  
6   Northern Metropolitan        4019.0  
7   Northern Metropolitan        4019.0  
9   Northern Metropolitan        4019.0  
11  Northern Metropolitan        4019.0  
12  Northern Metropolitan        4019.0  
15  Northern Metropolitan        4019.0  
16  Northern Metropolitan        4019.0  

[10 rows x 21 columns]

In [197]:
data.select_dtypes('object').head(4)

Suburb          Address Type Method SellerG       Date CouncilArea  \
1  Abbotsford  25 Bloomburg St    h      S  Biggin  4/02/2016       Yarra   
2  Abbotsford     5 Charles St    h     SP  Biggin  4/03/2017       Yarra   
4  Abbotsford      55a Park St    h     VB  Nelson  4/06/2016       Yarra   
6  Abbotsford     124 Yarra St    h      S  Nelson  7/05/2016       Yarra   

              Regionname  
1  Northern Metropolitan  
2  Northern Metropolitan  
4  Northern Metropolitan  
6  Northern Metropolitan

In [198]:
# convert the 'Date' column to datetime format
data['Date'] = data['Date'].astype('datetime64[ns]')

In [199]:
for col in data.select_dtypes('object').columns:
    if len(np.unique(data[col]))<10:
        print(col,data[col].value_counts(),sep='\n\n',end='\n\n')
    else:
        print(col,"Unique values : ",len(np.unique(data[col])),sep='\n',end='\n\n')

Suburb
Unique values : 
293

Address
Unique values : 
7353

Type

h    4727
u    2002
t     719
Name: Type, dtype: int64

Method

S     4780
SP    1034
PI     915
VB     672
SA      47
Name: Method, dtype: int64

SellerG
Unique values : 
216

CouncilArea
Unique values : 
31

Regionname

Southern Metropolitan         2711
Northern Metropolitan         2193
Western Metropolitan          1604
Eastern Metropolitan           696
South-Eastern Metropolitan     181
Eastern Victoria                26
Northern Victoria               20
Western Victoria                17
Name: Regionname, dtype: int64



In [200]:
data.dtypes.value_counts()

object            7
float64           7
int64             6
datetime64[ns]    1
dtype: int64

## Data normalization
#### Using Z- Score method (often called standardization) 
- Transforms the info into distribution with a mean of 0 and a typical deviation of 1. 
- Each standardized value is computed by subtracting the mean of the corresponding feature then dividing by the quality deviation.



In [202]:
for col in ['Rooms', 'Price', 'Distance', 'Bedroom2', 'Bathroom', 'Car','Landsize', 'BuildingArea','Propertycount']:
    data[col]=(data[col]-data[col].mean())/data[col].std()

In [203]:
data.head()

Suburb          Address     Rooms Type     Price Method SellerG  \
1  Abbotsford  25 Bloomburg St -0.906538    h -0.028352      S  Biggin   
2  Abbotsford     5 Charles St  0.113540    h  0.617465     SP  Biggin   
4  Abbotsford      55a Park St  1.133618    h  0.820221     VB  Nelson   
6  Abbotsford     124 Yarra St  0.113540    h  1.234745      S  Nelson   
7  Abbotsford    98 Charles St -0.906538    h  0.874290      S  Nelson   

        Date  Distance  Postcode  ...  Bathroom       Car  Landsize  \
1 2016-04-02 -1.285378      3067  ... -0.793999 -1.687216 -0.328018   
2 2017-04-03 -1.285378      3067  ...  0.631677 -1.687216 -0.350404   
4 2016-04-06 -1.285378      3067  ... -0.793999  0.476977 -0.364651   
6 2016-07-05 -1.285378      3067  ...  0.631677 -1.687216 -0.237454   
7 2016-08-10 -1.285378      3067  ... -0.793999  0.476977 -0.226260   

   BuildingArea  YearBuilt  CouncilArea Lattitude  Longtitude  \
1     -0.775467       1900        Yarra  -37.8079    144.9934   
2      0.080581       1900        Yarra  -37.8093    144.9944   
4     -0.015876       2014        Yarra  -37.8072    144.9941   
6      0.804001       1910        Yarra  -37.8024    144.9993   
7     -0.437871       1890        Yarra  -37.8060    144.9954   

              Regionname Propertycount  
1  Northern Metropolitan     -0.797246  
2  Northern Metropolitan     -0.797246  
4  Northern Metropolitan     -0.797246  
6  Northern Metropolitan     -0.797246  
7  Northern Metropolitan     -0.797246  

[5 rows x 21 columns]

# 4. Turn categorical variables into quantitative variables in Python. 

In [204]:
for col in data.select_dtypes(include='object'):
    data[col]=data[col].astype('category')
    data[col]=data[col].cat.codes

In [206]:
# One hot encoding strategy is to convert each category value into a new column and assigns a 1 or 0 
# (True/False) value to the column.

data=pd.get_dummies(data, columns=["Type"])
data.head()

Suburb  Address     Rooms     Price  Method  SellerG       Date  Distance  \
1       0     3323 -0.906538 -0.028352       1       19 2016-04-02 -1.285378   
2       0     5423  0.113540  0.617465       3       19 2017-04-03 -1.285378   
4       0     5879  1.133618  0.820221       4      120 2016-04-06 -1.285378   
6       0     1229  0.113540  1.234745       1      120 2016-07-05 -1.285378   
7       0     7335 -0.906538  0.874290       1      120 2016-08-10 -1.285378   

   Postcode  Bedroom2  ...  BuildingArea  YearBuilt  CouncilArea  Lattitude  \
1      3067 -0.884194  ...     -0.775467       1900           29   -37.8079   
2      3067  0.140783  ...      0.080581       1900           29   -37.8093   
4      3067  0.140783  ...     -0.015876       2014           29   -37.8072   
6      3067  1.165759  ...      0.804001       1910           29   -37.8024   
7      3067 -0.884194  ...     -0.437871       1890           29   -37.8060   

   Longtitude  Regionname  Propertycount  Type_0  Type_1  Type_2  
1    144.9934           2      -0.797246       1       0       0  
2    144.9944           2      -0.797246       1       0       0  
4    144.9941           2      -0.797246       1       0       0  
6    144.9993           2      -0.797246       1       0       0  
7    144.9954           2      -0.797246       1       0       0  

[5 rows x 23 columns]

In [207]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7448 entries, 1 to 12212
Data columns (total 23 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Suburb         7448 non-null   int16         
 1   Address        7448 non-null   int16         
 2   Rooms          7448 non-null   float64       
 3   Price          7448 non-null   float64       
 4   Method         7448 non-null   int8          
 5   SellerG        7448 non-null   int16         
 6   Date           7448 non-null   datetime64[ns]
 7   Distance       7448 non-null   float64       
 8   Postcode       7448 non-null   int64         
 9   Bedroom2       7448 non-null   float64       
 10  Bathroom       7448 non-null   float64       
 11  Car            7448 non-null   float64       
 12  Landsize       7448 non-null   float64       
 13  BuildingArea   7448 non-null   float64       
 14  YearBuilt      7448 non-null   int64         
 15  CouncilArea    7448 